src/Artifactory/ArtifactoryDownloadClient.ts

```diff
@@ -11,6 +18,56 @@ export class ArtifactoryDownloadClient {
             method: 'GET',
             headers: { Connection: 'Keep-Alive' },
         };
-        return await this.httpClient.doAuthRequest(requestParams);
+        return (await this.httpClient.doAuthRequest(requestParams)).data;
+    }
+
+    public async downloadArtifactToFile(from: string, to: string): Promise<void> {
+        this.logger.debug('Sending download artifact request...');
+        const requestParams: IRequestParams = {
+            url: encodeURI(from),
+            method: 'GET',
+            responseType: 'stream',
+        };
+        const writer: fs.WriteStream = fs.createWriteStream(to, { flags: 'wx' });
+        return new Promise((resolve, reject) => {
+            this.httpClient.doAuthRequest(requestParams).then((response) => {
+                if (response.status === 200) {
+                    response.data.pipe(writer);
+                } else {
+                    writer.close();
+                    reject(`Server responded with ${response.status}: ${response.data}`);
+                }
+            });
+            writer.on('end', () => {
+                console.log('download from' + from + ' to ' + to + ' was successful');
+                resolve();
+            });
```

Comment:
I think that 'end' event is only fired on read streams:
https://stackoverflow.com/a/34310963/4233342
```suggestion
```